Jetchat/app/src/main/java/com/example/compose/jetchat/data/Tokenizer.kt

```diff
@@ -22,5 +22,21 @@ class Tokenizer {
 
             return tokens
         }
+
+        /**
+         * Trim the input text to be under the number of tokens specified
+         *
+         * @return substring of the text that's no longer than the
+         * specified number of tokens
+         */
+        fun trimToTokenLimit (text: String?, tokenLimit: Int): String? {
+            // TODO: limit by tokens instead of the rough character approximation
+            val charLimit = tokenLimit * 4
+            return if (text?.length!! <= charLimit) {
```

Comment:
```suggestion
            val length = text?.length
            return if (length != null && length <= charLimit) {
```

Update logic to allow for possibly null text field (`!!` assertion will throw exception)